pkg/ast/ast_node.go

```diff
@@ -70,6 +70,11 @@ func (n Node) NameString(definition *Document) string {
 	return unsafebytes.BytesToString(definition.NodeNameBytes(n))
 }
 
+func (d *Document) UpdateRootNode(ref int, newRef int, newKind NodeKind) {
```

Comment:
instead of newRef,newKind it should be newNodeRef,newNodeKind to distinguish between root node (ref) and Node